### ISSUE TITLE
`tstr` type allows maps?

### ISSUE THREAD

**Issue by adzialocha** [anweiss/cddl#82]
Thank you for this fantastic implementation of CDDL! :sparkles: 

I've stumbled upon the following issue and wonder if it might be a bug?
The following CDDL schema (wrongly?) allows maps as values in the `fields` map even though only `tstr` types are permitted:

```
message = {
    fields: {
        + tstr => tstr
    }
}
```

This CBOR here (with diagnostic JSON) gets accepted by this schema while I would expect an error:

```
A1666669656C6473A16474657374A164546578746F48656C6C6F2C204D65737361676521
```

```json
{
  "fields": {
    "test": {
      "Text": "Hello, Message"
    }
  }
}
```

**Comment by anweiss** [comment]
Good catch, thanks for reporting @adzialocha! Fixing in #85.

**Comment by adzialocha** [comment]
Thank you for this fast fix!

**Comment by anweiss** [comment]
This has been fixed as of [v0.8.6](https://github.com/anweiss/cddl/releases/tag/0.8.6)

**Comment by adzialocha** [comment]
This might still be a problem @anweiss as maps are still considered valid strings in this test using version `v0.8.6.`. Maybe we're doing something wrong with the schema, but just to be sure I'm posting this here as well:

```rust
let schema = r#"
    message = {
        fields: {
            + tstr => tstr,
        }
    }
"#;

// {"fields": {"test":  "Hello, Message!"}}
let correct_cbor = "A1666669656C6473A164746573746F48656C6C6F2C204D65737361676521";

// {"fields": {"test": {"Text": "Hello, Message!"}}}
let invalid_cbor = "A1666669656C6473A16474657374A164546578746F48656C6C6F2C204D65737361676521";

assert!(cddl::validate_cbor_from_slice(schema, correct_cbor.as_bytes()).is_ok());
assert!(cddl::validate_cbor_from_slice(schema, invalid_cbor.as_bytes()).is_err());
```

**Comment by anweiss** [comment]
Hmmm ... I think you're right. Reopening to do some more troubleshooting here. Thanks for reporting.

**Comment by anweiss** [comment]
Hey @adzialocha this is (finally) being fixed in #79
